_posts/2023/2023-07-07-cobbler-inside-libvirt-1.md

```diff
@@ -0,0 +1,300 @@
+which is in the "cobbler" network. To achieve this edit the file `/etc/sysconfig/dhcpd` and change `DHCPD_INTERFACE=""` 
+to the name of your secondary interface. The entry should look similar to this afterwards:
+
+```shell
+DHCPD_INTERFACE="enp7s0"
+```
+
+Another thing that is missing for our secondary interface is its static configuration. Since we are hosting the DHCP 
+server, there is nothing assinging our interface its IP, gateway, and DNS server. As such we neeed to configure this 
+statically. openSUSE Tumbleweed uses NetworkManager per default to configure its networks.
+
+```shell
+nmcli con mod "Wired connection 1" ipv4.addresses "10.17.3.2/24"
+nmcli con mod "Wired connection 1" ipv4.gateway "10.17.3.1"
+nmcli con mod "Wired connection 1" ipv4.dns "8.8.8.8"
+nmcli con mod "Wired connection 1" ipv4.method "manual"
+nmcli c up "Wired connection 1"
+```
+
+> You will need to adjust the name of the interface `Wired Connection 1` to the interface that isn't online. This will
+> be the interface written in white normally. The other two interfaces should have a green tone.
+
+### Install & Configure Cobbler
+
+At the time of writing this guide, the current version of Cobbler in openSUSE Tumbleweed is 3.3.3. This can be 
+installed with:
+
+```shell
+zypper in cobbler
+```
+
+Confirm the installation with `y` when prompted and confirm by hitting the enter key. This will also install the TFTP 
+server along with several other required dependencies.
+
+In a default openSUSE Tumbleweed installation `firewalld` is configured and active. To open the required ports we 
+execute the following commands:
+
+```shell
+firewall-cmd --permanent --zone=public --add-service=http
+firewall-cmd --permanent --zone=public --add-service=https
+firewall-cmd --permanent --zone=public --add-service=tftp
+firewall-cmd --permanent --zone=public --add-service=dhcp
+firewall-cmd --permanent --zone=public --add-service=dhcpv6
+```
+
+> Be sure that this is executed on the guest. Opening these ports on the host may have unexpected consequences.
+
+Since now clients can access the desired services we should start them up.
+
+- Enable the TFTP service with: `systemctl enable --now tftp.service`
+- Enable the webserver with: `systemctl enable --now apache2.service`
+- Enable the DHCP service with: `systemctl enable --now dhcpd.service` - this will fail but this is expected at this 
+  point
+
+Now we can finally start configuring Cobbler itself. To do so edit the central configuration file at 
+`/etc/cobbler/settings.yaml` and make it match the IP on the guest (ignore the rest of the settings file for now):
+
+```yml
+server: 10.17.3.2
+next_server_v4: 10.17.3.2
+manage_dhcp: true
+manage_dhcp_v4: true
+```
+
+For this to take effect restart Cobbler:
+
+```shell
+systemctl restart cobblerd
+```
+
+Since Cobbler now knows its own IP address we can modify the DHCP template that Cobbler ships to match our requirements.
```

Comment:
Same as above but different solution. Again not wrong, but reads better in my view. feel free to disregard  though. This solution indicates the condition that leads to us being able to edit the file. (Yes, I love nitpicking about grammar though I am not a professional myself. :P)
```suggestion
Now that Cobbler knows its own IP address, we can modify the DHCP template that Cobbler ships to match our requirements.
```